set.mm

```diff
@@ -528868,6 +528934,404 @@ Set induction (or epsilon induction)
+     A. m e. n ( f ` m ) R ( f ` suc m ) ) } $.
+  $}
+
+  ${
+    $d R f m n x y $.  $d S f m n x y $.
+    $( Equality theorem for transitive closure.  (Contributed by Scott Fenton,
+       17-Oct-2024.) $)
+    ttrcleq $p |- ( R = S -> t++ R = t++ S ) $=
+      ( vf vn vx vy vm wceq csuc wfn cfv wbr wral w3a wex copab cttrcl df-ttrcl
+      cv wrex com c1o cdif breq ralbidv 3anbi3d exbidv rexbidv opabbidv 3eqtr4g
+      c0 wa ) ABHZCSZDSZIJZUKUNKESHUOUNKFSHULZGSZUNKZURIUNKZALZGUOMZNZCOZDUAUBU
+      CZTZEFPUPUQUSUTBLZGUOMZNZCOZDVETZEFPAQBQUMVFVKEFUMVDVJDVEUMVCVICUMVBVHUPU
+      QUMVAVGGUOUSUTABUDUEUFUGUHUIEFACGDREFBCGDRUJ $.
+  $}
+
+  ${
+    $d R f m n y z $.  $d x f m n y z $.
+    nfttrcl.1 $e |- F/_ x R $.
+    $( Bound variable hypothesis builder for transitive closure.  (Contributed
+       by Scott Fenton, 17-Oct-2024.) $)
+    nfttrcl $p |- F/_ x t++ R $=
+      ( vf vn vy vz vm cttrcl cv csuc wfn c0 cfv wceq wa wbr wral nfcv nfv cdif
+      w3a wex com c1o wrex copab df-ttrcl nfbr nfralw nf3an nfrex nfopab nfcxfr
+      nfex ) ABIDJZEJZKLZMUPNFJOUQUPNGJOPZHJZUPNZUTKUPNZBQZHUQRZUBZDUCZEUDUEUAZ
+      UFZFGUGFGBDHEUHVHFGAVFAEVGAVGSVEADURUSVDAURATUSATVCAHUQAUQSAVAVBBAVASCAVB
+      SUIUJUKUOULUMUN $.
+  $}
+
+  ${
+    $d R f n m x y $.
+    $( The transitive closure of a class is a relationship.  (Contributed by
+       Scott Fenton, 17-Oct-2024.) $)
+    relttrcl $p |- Rel t++ R $=
+      ( vf vn vx vy vm cv csuc wfn c0 cfv wceq wa wbr wral w3a wex com c1o cdif
+      wrex cttrcl df-ttrcl relopabi ) BGZCGZHIJUEKDGLUFUEKEGLMFGZUEKUGHUEKANFUF
+      OPBQCRSTUADEAUBDEABFCUCUD $.
+  $}
+
+  ${
+    $d A x y n f a $.  $d B x y n f a $.  $d R x y n f a $.
+    $( Binary relationship form of transitive closure.  (Contributed by Scott
+       Fenton, 18-Aug-2024.) $)
+    brttrcl $p |- ( A t++ R B <-> E. n e. ( _om \ 1o ) E. f
+     ( f Fn suc n /\ ( ( f ` (/) ) = A /\ ( f ` n ) = B ) /\
+     A. a e. n ( f ` a ) R ( f ` suc a ) ) ) $=
+      ( vx vy wbr cvv wcel wa cv csuc c0 cfv wceq w3a wex wrex wfn wral com c1o
+      cttrcl cdif relttrcl brrelex12i fvex eleq1 mpbii anim12i 3ad2ant2 exlimiv
+      rexlimivw eqeq2 anbi1d 3anbi2d exbidv rexbidv anbi2d df-ttrcl pm5.21nii
+      brabg ) ABCUEZIAJKZBJKZLZDMZEMZNUAZOVIPZAQZVJVIPZBQZLZFMZVIPVQNVIPCIFVJUB
+      ZRZDSZEUCUDUFZTZABVECUGUHVTVHEWAVSVHDVPVKVHVRVMVFVOVGVMVLJKVFOVIUIVLAJUJU
+      KVOVNJKVGVJVIUIVNBJUJUKULUMUNUOVKVLGMZQZVNHMZQZLZVRRZDSZEWATVKVMWFLZVRRZD
+      SZEWATWBGHABJJVEWCAQZWIWLEWAWMWHWKDWMWGWJVKVRWMWDVMWFWCAVLUPUQURUSUTWEBQZ
+      WLVTEWAWNWKVSDWNWJVPVKVRWNWFVOVMWEBVNUPVAURUSUTGHCDFEVBVDVC $.
+  $}
+
+  ${
+    $d A m n f a $.  $d B m n f a $.  $d R m n f a $.
+    $( Binary relationship form of transitive closure.  (Contributed by Scott
+       Fenton, 24-Aug-2024.) $)
+    brttrcl2 $p |- ( A t++ R B <-> E. n e. _om E. f
+     ( f Fn suc suc n /\ ( ( f ` (/) ) = A /\ ( f ` suc n ) = B ) /\
+     A. a e. suc n ( f ` a ) R ( f ` suc a ) ) ) $=
```

Comment:
Not sure it's useful to have it in addition to brttrcl.

Reply:
This is there so I can more easily use finds on transitive closures later on. Specifically, I use it in ttrclss.